source/fuzz/protobufs/spvtoolsfuzz.proto

```diff
@@ -517,6 +520,28 @@ message TransformationAddDeadContinue {
 
 }
 
+message TransformationAddFloatCastSynonym {
+
+  // Inserts instructions to the module that convert
+  // |synonym_id| to float and back, and adds a fact that
+  // the converted id and |synonym_id| are synonymous.
+
+  // Id to apply conversion to. This must be a result id of vector or integer type.
+  // If the type is vector, its components must have an integer type.
+  uint32 synonym_id = 1;
+
+  // Fresh id used to store the result of the conversion to float.
+  uint32 to_float_fresh_id = 2;
+
+  // Fresh id used to store the result of the conversion back to int.
+  uint32 to_int_fresh_id = 3;
+
+  // Required id of the float type to convert to. This can be a result id
+  // of either scalar or vector type, whose components are of type float.
+  uint32 float_type_id = 4;
```

Comment:
Do we need to pass this in to the transformation?

I think instead that the type of what you currently call synonym_id dictates the required float type, thus there is no need to pass it in to the transformation.  The transformation can instead check that an appropriate type exists in the module, and IsApplicable can return false if the right type does not exist.

Reply:
There is no function to check that a floating type exists in the module (something similar to `fuzzerutil::MaybeGetPointerType`). This is related to #3428.

Reply:
Moved this transformation into a separate PR #3472.